### ISSUE TITLE
 Could not find the correct Provider<(ReduxAction<dynamic>, {bool notify}) => void> above this ReduxSelector<AppState, dynamic> Widget

### ISSUE THREAD

**Issue by AnsarILeaf** [marcglasberg/provider_for_redux#6]
Following is error while using ReduxSelector

Error: Could not find the correct Provider<(ReduxAction<dynamic>, {bool notify}) => void> above this ReduxSelector<AppState, dynamic> Widget

This happens because you used a `BuildContext` that does not include the provider
of your choice. There are a few common scenarios:

- You added a new provider in your `main.dart` and performed a hot-reload.
  To fix, perform a hot-restart.

- The provider you are trying to read is in a different route.

  Providers are "scoped". So if you insert of provider inside a route, then
  other routes will not be able to access that provider.

- You used a `BuildContext` that is an ancestor of the provider you are trying to read.

  Make sure that ReduxSelector<AppState, dynamic> is under your MultiProvider/Provider<(ReduxAction<dynamic>, {bool notify}) => void>.
  This usually happens when you are creating a provider and trying to read it immediately.

  For example, instead of:

  ```
  Widget build(BuildContext context) {
    return Provider<Example>(
      create: (_) => Example(),
      // Will throw a ProviderNotFoundError, because `context` is associated
      // to the widget that is the parent of `Provider<Example>`
      child: Text(context.watch<Example>()),
    ),
  }
  ```

  consider using `builder` like so:

  ```
  Widget build(BuildContext context) {
    return Provider<Example>(
      create: (_) => Example(),
      // we use `builder` to obtain a new `BuildContext` that has access to the provider
      builder: (context) {
        // No longer throws
        return Text(context.watch<Example>()),
      }
    ),
  }
  ```

**Comment by savy-91** [comment]
Having the exact same issue

**Comment by savy-91** [comment]
If you pin to these versions, it works @AnsarILeaf 
```
  async_redux: 7.0.1
  provider_for_redux: 1.1.4
```

**Comment by AnsarILeaf** [comment]
> If you pin to these versions, it works @AnsarILeaf
> 
> ```
>   async_redux: 7.0.1
>   provider_for_redux: 1.1.4
> ```

Yes thanks @savy-91, i've already done that. But issue is with version 1.1.5

**Comment by marcglasberg** [comment]
Please see if the problem is solved for version: 2.0.0-nullsafety.2

**Comment by marcglasberg** [comment]
@AnsarILeaf @savy-91 Hello, could you please confirm if the problem is solved (or not)? I'm going to close this for the moment. Thanks.
